operators/src/engine/result_descriptor.rs

```diff
@@ -69,10 +71,25 @@ impl ResultDescriptor for RasterResultDescriptor {
 }
 
 /// A `ResultDescriptor` for vector queries
-#[derive(Debug, Copy, Clone, Ord, PartialOrd, Eq, PartialEq, Serialize, Deserialize)]
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
 pub struct VectorResultDescriptor {
     pub data_type: VectorDataType,
     pub spatial_reference: SpatialReferenceOption,
+    pub columns: HashMap<String, FeatureDataType>,
+}
+
+impl VectorResultDescriptor {
+    /// Create a new `VectorResultDescriptor` by only using new `columns`
+    pub fn map_columns<F>(&self, f: F) -> Self
```

Comment:
why not consume self like the other map functions on result descriptor?

Reply:
Because you now get result descriptors by reference and then you would always need to say clone and then map. Then you've cloned data and discard it, e.g. a hash map, so I think it makes sense this way.